File: core/inventory/tests/fixtures/sanitizer_cut_corpus_cpp/symmetric_sanitize.c
/* Symmetric sanitize across if-branches.
 *
 * Phase 11 verdict: suppress.
 *
 * Both branches produce the cleaned value into ``out``; control-flow
 * cut holds and value-bound conditions 2 and 3 fire on both branch's
 * binding. Lexical-only would miss this (no single sanitizer
 * dominates both branches in the legacy regex check). The value-
 * bound gate is the design's motivating case.
 */
extern char *g_markup_escape_text(const char *, long);
extern void render(const char *);

void handle(const char *x, int trusted) {
    const char *out;
    if (trusted) {
        out = g_markup_escape_text(x, -1);
    } else {
        out = g_markup_escape_text(x, -1);
    }
    render(out);
}

File: core/inventory/tests/fixtures/sanitizer_cut_corpus/symmetric_sanitize.py
"""Symmetric sanitize — both branches sanitize the right symbol.

Phase 4 verdict: suppress.

Every path crosses an html.escape on user.name; both bindings'
output_symbols={safe}; sink reads safe. All four conditions of the
gate hold.

The lexical check at smt_barrier.py:746 misses this case (sanitizer
in one branch doesn't lexically precede the sink — the if/else
structure has no single dominating line). Vertex-cut over the
union of both bindings correctly suppresses.
"""


def handle(user):
    if user.is_admin:
        safe = html.escape(user.name)  # noqa: F821 — fixture
    else:
        safe = html.escape(user.name)  # noqa: F821 — fixture
    render(safe)                       # noqa: F821 — fixture
